pkg/services/syncer/syncer.go

```diff
@@ -124,22 +124,20 @@ func syncNamespace(config config.Config, namespace string) error {
 		}
 	}
 
-	// TODO: collect all errors
 	for _, vv := range versions {
 		for _, v := range vv {
 			cli := config.Clients.OS.ManagedOSVersion()
 
 			_, err := cli.Get(namespace, v.ObjectMeta.Name, metav1.GetOptions{})
 			if err == nil {
-				logrus.Debugf("there is already a version defined for %s(%s)", v.Name, v.Spec.Version)
+				msg := fmt.Sprintf("there is already a version defined for %s(%s)", v.Name, v.Spec.Version)
+				config.Recorder.Event(&v, corev1.EventTypeWarning, "sync", msg)
```

Comment:
ditto here, `v` still doexn't exist at this stage. We have to report such failures back to the channel that originated it

Reply:
as far as I can see, the event uses the object reference to send the event in the same namespace, but that should sync with our current `v` object as they are the same object and should run in the same namespace no? Otherwise I have no idea how to report to the `upgradechannel` here

Reply:
sorry, I've misread the `err == nil` here :facepalm: looks good here

Reply:
well that is good, now Ifeel less alone in my confusion :P